pkgs/tools/filesystems/fatresize/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, parted, substituteAll, utillinux, ... }:
+
+stdenv.mkDerivation rec {
+  
+  version = "1.0.2";
+  name = "fatresize-${version}";
+
+  src = fetchurl {
+    url = mirror://sourceforge/fatresize/fatresize-1.0.2.tar.bz2;
+    sha256 = "04wp48jpdvq4nn0dgbw5za07g842rnxlh9mig4mslz70zqs0izjm";
+  };
+  
+  phases = [
+    "unpackPhase"
+    "patchPhase"
+    "configurePhase"
+    "buildPhase"
+    "installPhase"
+  ];
+  
+  buildInputs = [ parted utillinux ];
+  
+  patches = [ ./parted_nix.patch ];
+  
+  preConfigure = ''
+    echo "Replacing calls to ped_free with free ..."
+    substituteInPlace ./fatresize.c --replace ped_free free
+  '';
+  
+  makeFlags = ''
+    LDFLAGS=-lparted-fs-resize
+  '';
+  
+  propagatedBuildInputs = [ parted utillinux ];
+  
+  meta = {
+    description = "The FAT16/FAT32 non-destructive resizer.";
```

Comment:
The full stop is unnecessary. See http://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.